File: src/main/java/pw/switchcraft/plethora/integration/computercraft/meta/item/PocketComputerItemMeta.java
package pw.switchcraft.plethora.integration.computercraft.meta.item;

import dan200.computercraft.api.pocket.IPocketUpgrade;
import dan200.computercraft.shared.pocket.items.ItemPocketComputer;
import net.minecraft.item.ItemStack;
import pw.switchcraft.plethora.api.meta.ItemStackMetaProvider;

import javax.annotation.Nonnull;
import java.util.HashMap;
import java.util.Map;

public final class PocketComputerItemMeta extends ItemStackMetaProvider<ItemPocketComputer> {
    public PocketComputerItemMeta() {
        super("pocket", ItemPocketComputer.class);
    }

    @Nonnull
    @Override
    public Map<String, ?> getMeta(@Nonnull ItemStack stack, @Nonnull ItemPocketComputer pocket) {
        Map<String, Object> out = new HashMap<>(2);

        int colour = pocket.getColour(stack);
        if (colour != -1) {
            out.put("color", colour);
            out.put("colour", colour);
        }

        out.put("back", getUpgrade(ItemPocketComputer.getUpgrade(stack)));

        return out;
    }

    private static Map<String, String> getUpgrade(IPocketUpgrade upgrade) {
        if (upgrade == null) return null;

        Map<String, String> out = new HashMap<>(2);
        out.put("id", upgrade.getUpgradeID().toString());
        out.put("adjective", upgrade.getUnlocalisedAdjective());

        return out;
    }
}

File: src/main/java/io/sc3/plethora/integration/computercraft/meta/item/PocketComputerItemMeta.kt
package io.sc3.plethora.integration.computercraft.meta.item

import dan200.computercraft.api.pocket.IPocketUpgrade
import dan200.computercraft.shared.pocket.items.PocketComputerItem
import io.sc3.plethora.api.meta.ItemStackMetaProvider
import net.minecraft.item.ItemStack

class PocketComputerItemMeta : ItemStackMetaProvider<PocketComputerItem>(PocketComputerItem::class.java, "pocket") {
  override fun getMeta(stack: ItemStack, item: PocketComputerItem): Map<String, *> {
    val out: MutableMap<String, Any?> = HashMap(2)

    val colour = item.getColour(stack)
    if (colour != -1) {
      out["color"] = colour
      out["colour"] = colour
    }

    out["back"] = getUpgrade(PocketComputerItem.getUpgrade(stack))

    return out
  }

  companion object {
    private fun getUpgrade(upgrade: IPocketUpgrade?): Map<String, String>? {
      if (upgrade == null) return null
      return mapOf(
        "id"        to upgrade.upgradeID.toString(),
        "adjective" to upgrade.unlocalisedAdjective
      )
    }
  }
}
